content/source/docs/cloud/sentinel/sentinel-tf-012.html.md

```diff
@@ -170,6 +184,35 @@ namespace](./import/tfplan.html#namespace-resource-diff)
 to validate whether or not a value is unknown before looking for it in
 `applied`.
 
+### Changes Affecting Resources Being Destroyed but not Re-created
+
+In Terraform 0.11, when a resource is being destroyed but not re-created, it's
+[`diff`](./import/tfplan.html#value-diff) value in the `tfplan` import is empty.
+In Terraform 0.12, however, the `diff` value does has data. For both versions of
+Terraform, the [`applied`](./import/tfplan.html#value-applied) value will be
+absent.
+
+It is therefore recommended to check whether resources are being destroyed and
+not re-created in all Sentinel policies that use the `tfplan` import and the
+`applied` value in order to avoid `undefined` values in your functions and
+rules.
+
+Before Terraform 0.12 was released, some Sentinel policies tested whether
+a resource was being destroyed and not re-created by testing the condition
+`length(r.diff) == 0`. However, that cannot be used with Terraform 0.12.
+
+Fortunately, after Terraform 0.12 was released, new
+[`destroy`](./import/tfplan.html#value-destroy) and
+[`requires_new`](./import/tfplan.html#value-requires_new) values were added to
+the `tfplan` import. Since these values are available both for Terraform
+0.11 and 0.12, you can now simply test `r.destroy and not r.requires_new` to
+see if a resource is being destroyed and not re-created with both versions of
+Terraform.
+
+Please note that if you are using Terraform Enterprise,
+you must use version v201909-1 or higher in order to use the `destroy` and
+`requires_new` values.
```

Comment:
Suggested edit:

> absent. This also means previous policies that tested whether a resource was being destroyed and not re-created by testing the condition `length(r.diff) == 0` will need to be updated.
>
>New [`destroy`](./import/tfplan.html#value-destroy) and [`requires_new`](./import/tfplan.html#value-requires_new) values have been added to the `tfplan` import. Since these values are available both for Terraform 0.11 and 0.12, you can now test `r.destroy and not r.requires_new` to see if a resource is being destroyed and not re-created with both versions of Terraform.
>
>It is therefore recommended to update all policies, in both Terraform 0.11 and 0.12, that use the `tfplan` import and the `applied` value to check whether resources are being destroyed and not re-created. This will avoid `undefined` values in your functions and rules.
>
>Please note that if you are using Terraform Enterprise, you must use version v201909-1 or higher in order to use the `destroy` and `requires_new` values.

Reply:
Thanks @glenngillen .  I revised the passage to include your suggestions, although I did not word things exactly as you did.  A new thing I mentioned which I learned about yesterday in discussions with @vancluever is that we recently made a change to the `tfplan` import so that the `applied` value is not present when a resource is being destroyed but not re-created.  Since that can affect existing policies, I thought it important to point that out.  I also made the doc's use of ticks more consistent and removed some excessive hyperlinks.